### PR TITLE
[NEXUS-8134] CMA Repository Management Security

### PR DESCRIPTION
**Pull request by adreghiciu**
https://issues.sonatype.org/browse/NEXUS-8134
http://bamboo.s/browse/NX3-OSSF215

### CODE REVIEW

File: plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/RepositoryComponent.groovy [hunk 1]
```diff
@@ -12,14 +12,13 @@
...
 package org.sonatype.nexus.coreui
 
-import javax.annotation.Nullable
-import javax.inject.Inject
-import javax.inject.Named
-import javax.inject.Singleton
-import javax.validation.Valid
-import javax.validation.constraints.NotNull
-import javax.validation.groups.Default
-
```
Comment (jdillon) [inline plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/RepositoryComponent.groovy]: looks like you also have wrong import order for groovy files:

https://github.com/sonatype/codestyle#imports

**Comment by adreghiciu** [comment]
CI :+1: http://bamboo.s/browse/NX3-OSSF215-1

File: plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/RepositoryComponent.groovy [hunk 2]
```diff
@@ -168,16 +178,27 @@ extends DirectComponentSupport
...
   @PackageScope
-  List<Repository> filter(final @Nullable StoreLoadParameters parameters) {
+  Iterable<Repository> filter(final @Nullable StoreLoadParameters parameters) {
     def repositories = repositoryManager.browse()
     if (parameters) {
       String format = parameters.getFilter('format')
       if (format) {
-        return repositories.findResults { Repository repository ->
-          repository.format == format
+        repositories = repositories.findResults { Repository repository ->
```
Comment (kellyrob99) [inline plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/RepositoryComponent.groovy]: Hmmm, forgot about this idiosyncrasy of findResults. Might be better off with just findAll which is a truer filter method

File: plugins/rapture/nexus-rapture-plugin/src/main/java/org/sonatype/nexus/rapture/internal/ui/PermissionXO.groovy [hunk 3]
```diff
@@ -13,6 +13,7 @@
 package org.sonatype.nexus.rapture.internal.ui
 
 import groovy.transform.ToString
+import org.apache.shiro.authz.permission.WildcardPermission
```
Comment (kellyrob99) [inline plugins/rapture/nexus-rapture-plugin/src/main/java/org/sonatype/nexus/rapture/internal/ui/PermissionXO.groovy]: something missing in this class?

**Comment by kellyrob99** [comment]
+1 in general, couple of minor points. Notably tho, we're absent any new tests and none of this appears to have touched on existing tests?

**Comment by adreghiciu** [comment]
There is not too much to be automated tested since we do not test any of the UI *Components

**Comment by mrprescott** [comment]
+1

**Comment by adreghiciu** [comment]
Manually merged https://github.com/sonatype/nexus-oss/commit/dc60c121cfd68f5f93dfd356df97b272b3b222bf

**Pull request closed without merging**